reconcile/test/fleet_labeler/test_fleet_labeler_cluster_inventory_lifecycle.py

```diff
@@ -0,0 +1,197 @@
+from collections.abc import Callable
+
+from reconcile.fleet_labeler.dependencies import Dependencies
+from reconcile.fleet_labeler.integration import (
+    FleetLabelerIntegration,
+)
+from reconcile.gql_definitions.fleet_labeler.fleet_labels import (
+    FleetLabelsSpecV1,
+)
+from reconcile.test.fleet_labeler.fixtures import (
+    build_cluster,
+    build_ocm_client,
+    build_vcs,
+    get_fixture_content,
+    label_spec_data_from_fixture,
+)
+
+
+def test_add_new_cluster(
+    integration: FleetLabelerIntegration,
+    dependencies: Dependencies,
+    gql_class_factory: Callable[..., FleetLabelsSpecV1],
+) -> None:
+    """
+    We have 0 clusters in the current inventory.
+    OCM API returns 1 cluster.
+
+    We expect an MR that desires 1 cluster in the inventory.
+    """
+    dependencies.vcs = build_vcs(content=get_fixture_content("0_clusters.yaml"))
+    spec = gql_class_factory(
+        FleetLabelsSpecV1, label_spec_data_from_fixture("0_clusters.yaml")
+    )
+    dependencies.label_specs_by_name = {spec.name: spec}
+    dependencies.ocm_clients_by_label_spec_name = {
+        spec.name: build_ocm_client(
+            discover_clusters_by_labels=[
+                build_cluster(
+                    name="cluster_name",
+                    cluster_id="cluster_id",
+                    subscription_labels={"sre-capabilities.dtp.managed-labels": "true"},
+                ),
+            ],
+        )
+    }
+
+    integration.reconcile(dependencies=dependencies)
+
+    dependencies.vcs.open_merge_request.assert_called_once_with(  # type: ignore[attr-defined]
```

Comment:
to avoid type ignore, you can make `build_vcs` return type as `Mock`, then save it as local var, then verify with `vcs.xxxx.asssert`